src/shared_definitions.hpp

```diff
@@ -304,20 +311,21 @@ struct index_file_options
 /* actual index */
 template <DbIndexType           dbIndexType,
           AlphabetEnum          origAlph,
-          AlphabetEnum          redAlph>    // <- all members of index_file_options that influence types
+          AlphabetEnum          redAlph,
+          bool                  fwd>    // <- all members of index_file_options that influence types
 struct index_file
```

Comment:
I don't think you should have two instances of `index_file`, instead you should have one `index_file` with two `index` members. The other things in the index should not be duplicated, right?